includes/Modules/Reader_Revenue_Manager.php

```diff
@@ -434,43 +474,86 @@ protected function setup_assets() {
 			new Script(
 				'googlesitekit-modules-reader-revenue-manager',
 				array(
-					'src'          => $base_url . 'js/googlesitekit-modules-reader-revenue-manager.js',
-					'dependencies' => array(
+					'src'           => $base_url . 'js/googlesitekit-modules-reader-revenue-manager.js',
+					'dependencies'  => array(
 						'googlesitekit-vendor',
 						'googlesitekit-api',
 						'googlesitekit-data',
 						'googlesitekit-modules',
+						'googlesitekit-notifications',
 						'googlesitekit-datastore-site',
 						'googlesitekit-datastore-user',
 						'googlesitekit-components',
 					),
+					'load_contexts' => array( Asset::CONTEXT_ADMIN_SITEKIT, Asset::CONTEXT_ADMIN_POST_EDITOR ),
 				)
 			),
 		);
 
-		if ( Feature_Flags::enabled( 'rrmModuleV2' ) ) {
+		if ( Feature_Flags::enabled( 'rrmModuleV2' ) && Block_Support::has_block_support() ) {
```

Comment:
I've added a check for WP version 5.8+ around the block editor asset initialisation. This is with the expectancy that we'll go ahead with this change as per the conversation on [Slack](https://fueled10up.slack.com/archives/C077DB69US2/p1740561723845019?thread_ts=1740542414.593299&cid=C077DB69US2).

Cc @ankitrox, you may want to bring this into the [PR](https://github.com/google/site-kit-wp/pull/10246/files) for https://github.com/google/site-kit-wp/issues/9962.

Reply:
Update: We've now agreed to go ahead with making this as per https://github.com/google/site-kit-wp/issues/9962#issuecomment-2689669224.

Reply:
@techanvil As you have already wrapped the condition around asset initialisation for block edit plugin for RRM, I believe we don't need a separate followup PR to do the same thing as it will take effect when this PR is merged.

Please let me know if this is fine or you want me to create a follow-up PR for #9962?

Reply:
CC: @wpdarren

Reply:
@ankitrox I believe this change should be sufficient and you shouldn't need to create a follow-up PR, unless Tom disagrees, thanks!

Reply:
Thanks @ankitrox and @nfmohit. That sounds sensible to me, let's keep the change scoped to this PR and not create a followup for 9962.